qcodes/instrument_drivers/american_magnetics/AMI430.py

```diff
@@ -490,11 +494,15 @@ def __init__(self, name, instrument_x, instrument_y,
         self._instrument_y = instrument_y
         self._instrument_z = instrument_z
 
-        if repr(field_limit).isnumeric() or isinstance(field_limit, collections.abc.Iterable):
+        self._field_limit: Union[float, Iterable[Callable]]
+        if isinstance(field_limit, collections.abc.Iterable):
             self._field_limit = field_limit
+        elif isinstance(field_limit, numbers.Real):
```

Comment:
Does this evaluate to True for numpy float/ints

Reply:
yes, for all the "real" types mentioned in qcodes.utils.types this evaluates to True (tested, not just claiming).